File: php/features.php
<?php
declare(strict_types=1);

// Acousticbrainz SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class AcousticbrainzFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new AcousticbrainzBaseFeature();
            case "test":
                return new AcousticbrainzTestFeature();
            default:
                return new AcousticbrainzBaseFeature();
        }
    }
}

File: py/features.py
# Acousticbrainz SDK feature factory

from feature.base_feature import AcousticbrainzBaseFeature
from feature.test_feature import AcousticbrainzTestFeature


def _make_feature(name):
    features = {
        "base": lambda: AcousticbrainzBaseFeature(),
        "test": lambda: AcousticbrainzTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
